>From 3ae05ed82e0d21297cc98dfaeff9acfa9549995b Mon Sep 17 00:00:00 2001
From: Charles Plessy <plessy@debian.org>
Date: Mon, 13 Aug 2012 08:58:49 +0900
Subject: [PATCH] Do not call update-mime and update-mime-database in maint.
 scripts as they are triggered.

---
 autoscripts/postinst-mime           |    3 ---
 autoscripts/postinst-sharedmimeinfo |    3 ---
 autoscripts/postrm-mime             |    1 -
 autoscripts/postrm-sharedmimeinfo   |    3 ---
 dh_installmime                      |   35 +----------------------------------
 5 files changed, 1 insertion(+), 44 deletions(-)
 delete mode 100644 autoscripts/postinst-mime
 delete mode 100644 autoscripts/postinst-sharedmimeinfo
 delete mode 100644 autoscripts/postrm-mime
 delete mode 100644 autoscripts/postrm-sharedmimeinfo

diff --git a/autoscripts/postinst-mime b/autoscripts/postinst-mime
deleted file mode 100644
index 3749fc9..0000000
--- a/autoscripts/postinst-mime
+++ /dev/null
@@ -1,3 +0,0 @@
-if [ "$1" = "configure" ] && [ -x "`which update-mime 2>/dev/null`" ]; then
-	update-mime
-fi
diff --git a/autoscripts/postinst-sharedmimeinfo b/autoscripts/postinst-sharedmimeinfo
deleted file mode 100644
index aa6333f..0000000
--- a/autoscripts/postinst-sharedmimeinfo
+++ /dev/null
@@ -1,3 +0,0 @@
-if [ "$1" = "configure" ] && [ -x "`which update-mime-database 2>/dev/null`" ]; then
-	update-mime-database /usr/share/mime
-fi
diff --git a/autoscripts/postrm-mime b/autoscripts/postrm-mime
deleted file mode 100644
index a940411..0000000
--- a/autoscripts/postrm-mime
+++ /dev/null
@@ -1 +0,0 @@
-if which update-mime >/dev/null 2>&1; then update-mime; fi
diff --git a/autoscripts/postrm-sharedmimeinfo b/autoscripts/postrm-sharedmimeinfo
deleted file mode 100644
index 917fd49..0000000
--- a/autoscripts/postrm-sharedmimeinfo
+++ /dev/null
@@ -1,3 +0,0 @@
-if [ -x "`which update-mime-database 2>/dev/null`" ]; then
-	update-mime-database /usr/share/mime
-fi
diff --git a/dh_installmime b/dh_installmime
index 998d6ba..d44c915 100755
--- a/dh_installmime
+++ b/dh_installmime
@@ -11,17 +11,13 @@ use Debian::Debhelper::Dh_Lib;
 
 =head1 SYNOPSIS
 
-B<dh_installmime> [S<I<debhelper options>>] [B<-n>]
+B<dh_installmime> [S<I<debhelper options>>]
 
 =head1 DESCRIPTION
 
 B<dh_installmime> is a debhelper program that is responsible for installing
 mime files into package build directories.
 
-It also automatically generates the F<postinst> and F<postrm> commands needed to
-interface with the debian B<mime-support> and B<shared-mime-info> packages. These
-commands are inserted into the maintainer scripts by L<dh_installdeb(1)>.
-
 =head1 FILES
 
 =over 4
@@ -38,22 +34,6 @@ directory.
 
 =back
 
-=head1 OPTIONS
-
-=over 4
-
-=item B<-n>, B<--noscripts>
-
-Do not modify F<postinst>/F<postrm> scripts.
-
-=back
-
-=head1 NOTES
-
-Note that this command is not idempotent. L<dh_prep(1)> should be called
-between invocations of this command. Otherwise, it may cause multiple
-instances of the same text to be added to maintainer scripts.
-
 =cut
 
 init();
@@ -68,12 +48,6 @@ foreach my $package (@{$dh{DOPACKAGES}}) {
 		}
 		doit("install","-p","-m644",$mime,"$tmp/usr/lib/mime/packages/$package");
 	}
-	# Check whether we have to call update-mime (either upstream already
-	# installs a MIME information file or Debian provides one)
-	if ((! $dh{NOSCRIPTS}) && (-d "$tmp/usr/lib/mime/packages")) {
-		autoscript($package,"postinst","postinst-mime");
-		autoscript($package,"postrm","postrm-mime")
-	}
 	
 	my $sharedmimeinfo=pkgfile($package,"sharedmimeinfo");
 	
@@ -83,13 +57,6 @@ foreach my $package (@{$dh{DOPACKAGES}}) {
 		}
 		doit("install", "-p", "-m644", $sharedmimeinfo, "$tmp/usr/share/mime/packages/$package.xml");
 	}
-	# check whether we have to call update-mime-database (either upstream
-	# already installs a shared MIME information file or Debian provides
-	# one)
-	if ((! $dh{NOSCRIPTS}) && (-d "$tmp/usr/share/mime/packages")) {
-		autoscript($package, "postinst", "postinst-sharedmimeinfo");
-		autoscript($package, "postrm", "postrm-sharedmimeinfo")
-	}
 }
 
 =head1 SEE ALSO
-- 
1.7.10

